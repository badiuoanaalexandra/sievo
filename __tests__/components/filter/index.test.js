import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import Filter from '../../../src/components/filter/index';

const mockStore = configureStore();

const initialState = {
  projectsReducer: [
      {"project":2, "description":"test", "start date": "2013-12-28T00:00:00+00:00", "savings amount":3722.41684, "currency":"NULL"},
      {"project":1, "description":"test 2", "start date": "2014-01-28T00:00:00+00:00", "savings amount":372.32, "currency":"EUR"}
  ],
  displayProjectsReducer:{changed:false, description:"", sortingOrder:"", changedProjects:null},
  uiReducer:1024
};

describe('<Filter />', () => {

  test("renders", () => {
    const store = mockStore(initialState);
    const wrapper = shallow(<Filter store={store} />);
    const component = wrapper.dive();
    expect(toJson(component)).toMatchSnapshot();
  });

  test("successfully triggers reset", () => {
    const store = mockStore(initialState);
    const wrapper = shallow(<Filter store={store}/>);
    const component = wrapper.dive();

    component.find("button.reset").simulate("click");
    const actions = store.getActions();
    expect(actions).toEqual([ { type: 'REQUEST_PROJECTS_RESET' } ]);
  });

  test("successfully triggers sort on mobile", () => {
    const store = mockStore(initialState);
    global.innerWidth = 500;
    const wrapper = shallow(<Filter store={store}/>);
    const component = wrapper.dive();

    component.find("button.sort").simulate("click");
    const actions = store.getActions();
    expect(actions).toEqual([{ type: 'REQUEST_PROJECTS_SORT',
      changedProjects: [
          {"project":2, "description":"test", "start date": "2013-12-28T00:00:00+00:00", "savings amount":3722.41684, "currency":"NULL"},
          {"project":1, "description":"test 2", "start date": "2014-01-28T00:00:00+00:00", "savings amount":372.32, "currency":"EUR"}
      ],
      description:"",
      sortField:"project",
      sortingOrder:""
    } ]);
  });

  test("successfully requests filtering", () => {
    const store = mockStore(initialState);
    const wrapper = shallow(<Filter store={store}/>);
    const component = wrapper.dive();

    component.find(".filter").find("input").simulate("change", {target: {value:"test 2"}});
    const actions = store.getActions();
    expect(actions).toEqual([{ type: 'REQUEST_PROJECTS_FILTER',
      filterProjects: [
          {"project":2, "description":"test", "start date": "2013-12-28T00:00:00+00:00", "savings amount":3722.41684, "currency":"NULL"},
          {"project":1, "description":"test 2", "start date": "2014-01-28T00:00:00+00:00", "savings amount":372.32, "currency":"EUR"}
      ],
      description:"test 2",
      descriptionField:"description",
      sortField:undefined,
      sortingOrder:""
    } ]);
  });

});
