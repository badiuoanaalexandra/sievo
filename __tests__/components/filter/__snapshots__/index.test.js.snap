// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> renders 1`] = `
<div
  className="filter-wrapper"
>
  <h1>
    Projects
  </h1>
  <div
    className="filter"
  >
    <input
      onChange={[Function]}
      placeholder="Enter description"
      type="text"
      value=""
    />
  </div>
  <a
    className=" sort button"
    onClick={[Function]}
  >
    SORT
  </a>
  <a
    className="reset button"
    onClick={[Function]}
  >
    RESET
  </a>
</div>
`;
